packages/core/src/fhirpath/parse.ts

```diff
@@ -248,8 +248,19 @@ export function evalFhirPath(expression: string, input: unknown): unknown[] {
  * Evaluates a FHIRPath expression against a resource or other object.
  * @param expression The FHIRPath expression to parse.
  * @param input The resource or object to evaluate the expression against.
+ * @param context The context to use when evaluating the expression.
  * @returns The result of the FHIRPath expression against the resource or object.
  */
-export function evalFhirPathTyped(expression: string, input: TypedValue[]): TypedValue[] {
-  return parseFhirPath(expression).eval(input);
+export function evalFhirPathTyped(
+  expression: string,
+  input: TypedValue[],
+  context: AtomContext | undefined = undefined
```

Comment:
Since this is the public interface, we might want to make it clearer what the user should pass in.  For example, maybe instead of asking them to construct an `AtomContext` (which is a purely internal concept), we could have them pass in a `Record<string, TypedValue>` for the variables

Reply:
Yeah, that's a good call.  You could go further and say there should be `interface FhirPathContext extends AtomContext` or something.

Also @jamestouri you can use the shorter syntax of `context?: AtomContext` for an optional arg.